### ISSUE TITLE
OTP 18.0 - "EDoc could not process file"

### ISSUE THREAD

**Issue by jfacorro** [erldocs/erldocs#25]
This [commit](https://github.com/erlang/otp/commit/6f492b98505ee6d68b9f438915dfbd3616a1b729#diff-b573efe5351c32eaaae32fb7c89cadccL988) in OTP introduced a bunch of changes to `EDoc`, one of which is breaking erldocs when it [calls](https://github.com/erldocs/erldocs/blob/master/src/specs_gen__R15_and_above.erl#L103) the `edoc_lib:get_env_doc/4` which no longer exists.

There might have been some other breaking change recently because I'm also getting an error when the `docgen_edoc_xml_cb` layout is applied on a source file that has a type defined in terms of a map.

```
edoc: error in layout 'docgen_edoc_xml_cb': {'EXIT',
                                           {function_clause,
                                            [{docgen_edoc_xml_custom,t_type,
                                              [[{xmlElement,type,[],[],
                                                 {xmlNamespace,[],[]},
                                                 [{map_field,1},
```

It seems the type of xmlElement is not being handled by the layout.

**Comment by fenollp** [comment]
Hey @jfacorro! Thanks for noticing!

(I did not update otp in the production server of erldocs.com yet. Still running 17.4 I think)

It seems like it's a simple fix: could you remove one of these empty lists and make a pull request?
Thanks!

**Comment by jfacorro** [comment]
@fenollp I think just removing an argument to the call of `edoc_lib:get_env_doc/3`  would break previous OTP versions, since the function with arity 4 was removed and replaced by the one with arity 3. Maybe it is necessary to make some sort of version checking in order to use one or the other?

**Comment by fenollp** [comment]
More like a `case` on `erlang:function_exported(edoc_lib, get_doc_env, 4)` ;)

**Comment by fenollp** [comment]
Fixed by #31

**Comment by NobbZ** [comment]
I did a fresh compile from master, but get exactly the above described error as far as I can see, I copied the last stacktrace into a [gist](https://gist.github.com/NobbZ/ad7a90904d63b43b49d6). This example is from trying to document `erldocs` itself after my own project didn't work either.

I'm on OTP 18.2 (installed from erlang-solutions.com's debian repository). And `erldocs` was compiled from exactly commit hash f042748c6d00efff2d3d2e353372560d39819c31
